cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/store/DatasetBasedTimeScheduleStore.java

```diff
@@ -476,7 +566,7 @@ private TriggerStatusV2 addAppVersion(TriggerStatusV2 oldTrigger) {
     return oldTrigger;
   }
 
-  private void upgradeJobs() throws Exception {
+  private void upgradeJobs(Table table) {
```

Comment:
also need to check whether `jobDetail` after upgrade already exists, don't add persist `jobDetail` if it already exists